modules/crypt.cpp

```diff
@@ -200,6 +332,41 @@ class CCryptMod : public CModule {
     }
 
     EModRet OnPrivNotice(CNick& Nick, CString& sMessage) override {
+        CString sCommand = sMessage.Token(0);
+        CString sOtherPubKey = sMessage.Token(1);
+
+        if ((sCommand.Equals("DH1080_INIT") || sCommand.Equals("DH1080_INIT_CBC")) && !sOtherPubKey.empty()) {
```

Comment:
History is like this:
- first there was DH1080_INIT and the keys generated with this were used for blowfish in ECB mode.
- as this ECB was deemed to insecure (always the same encoded word for the same unencoded word) a transition was made to use CBC.
- Mircryption started using this with DH1080_INIT_CBC.
- fish10 (and/or others) kept using DH1080_INIT but added a " CBC" suffix after the pubkey.
- KVIrc (and/or others) use CBC by default for all.

So DH1080_INIT and DH1080_INIT_CBC (and "DH1080_INIT pubkey CBC") all use the same method, but just differ in setting the secretkey: for CBC they mostly get a "cbc:" prefix, ECB gets nothing or "old:" or "ecb:". Then the client knows what decryption and encryption scheme to use, though most have fallback to either one depending on the received input (ECB encoded messages start with "+OK ", CBC with "+OK *").

As ECB is more insecure than CBC and thus to push CBC usage, I've chosen to use CBC by default even for the original DH1080_INIT, also because the crypt module only supports CBC mode.